wtxmgr/unconfirmed.go

```diff
@@ -53,25 +53,31 @@ func (s *Store) removeDoubleSpends(ns walletdb.ReadWriteBucket, rec *TxRecord) e
 	for _, input := range rec.MsgTx.TxIn {
 		prevOut := &input.PreviousOutPoint
 		prevOutKey := canonicalOutPoint(&prevOut.Hash, prevOut.Index)
-		doubleSpendHash := existsRawUnminedInput(ns, prevOutKey)
-		if doubleSpendHash != nil {
+
+		doubleSpendHashes := fetchUnminedInputSpendTxHashes(ns, prevOutKey)
+		for _, doubleSpendHash := range doubleSpendHashes {
+			doubleSpendVal := existsRawUnmined(ns, doubleSpendHash[:])
+			if doubleSpendVal == nil {
+				continue
```

Comment:
Is this an error case?

Reply:
Nope, this can happen in the case that the spending transaction actually spends multiple outputs from the same transaction. If this is the case, they'll get added multiple times, so it's possible to not find the unconfirmed transaction in the store if it's already been deleted. I added a comment to explain this. I tried to work around it but things got ugly quickly.

Reply:
Similar comment here about updating the comment to reflect reality.